storage/Devices/LuksImpl.cc

```diff
@@ -96,6 +107,14 @@ namespace storage
 	calculate_region_and_topology();
     }
 
+    void
+    Luks::Impl::set_integrity(const string& integrity)
+    {
+	Encryption::Impl::set_integrity(integrity);
+
+	if (get_integrity() != integrity)
```

Comment:
Is that ever the case?

Reply:
In reality no, but maybe someone put manually this suffix?

Reply:
After ```set_integrity(integrity)``` the condition is always false AFAIS.

Reply:
Ah right, this will never call the reize. I will move the condition before.

Reply:
Done. If you think that the condition should be dropped, we need to figure out why the tests are falling.

Reply:
Likely a check like in ```Md::Impl::calculate_region_and_topology()``` is needed:

```
	if (exists_in_system())
	    return;
```

Explanation is in Md::Impl.

Reply:
Added check in local `calculate_region_and_topology()`